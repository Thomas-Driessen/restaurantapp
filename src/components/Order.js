import React from 'react'
import Grid from '@material-ui/core/Grid';

class Order extends React.Component {
    componentDidMount() {
        fetch(`/api/order/${this.props.order}`)
        .then(res => res.json())
        .then((data) => {
          this.props.orderItems = data;
        })
        .catch(console.log)
    }
    render(){
    return(
        <div>
            { this.props.orderItems ? (    
            <Grid container spacing={24} style={{padding: 15}}>
                { this.props.orderItems.map(currentItem => (
                <Grid item xs={12} sm={6} lg={4} xl={3}>
                    <p>{currentItem.title}  {currentItem.price}</p>
                </Grid>
                ))}
            </Grid>
            ) : null}
        </div>
    )}
}
export default Order