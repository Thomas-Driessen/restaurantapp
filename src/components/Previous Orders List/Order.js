import React from 'react';
import Card from '@material-ui/core/Card';
import Typography from '@material-ui/core/Typography';

class Order extends React.Component {
    constructor(){
        super();
        this.state = {
            drink: []
        }
    }
    componentDidMount() {
        let mounted = true;
        fetch(`/api/${this.props.productType}/${this.props.orderId}`)
        .then(res => res.json())
        .then((data) => {
            if(mounted){
                this.setState({ drink: data })
            }
        })
        .catch(console.log)

        return() => mounted = false;
    }
    render(){
    return(
        <div>
            { this.state.drink ? (    
                <Card>
                    <Typography gutterBottom variant="inherit" component="h2">
                        {this.state.drink.title}
                    </Typography>
                    <Typography gutterBottom variant="inherit" component="h2" align="right">
                        {this.state.drink.price}â‚¬
                    </Typography>
                </Card>
            ) : null}
        </div>
    )}
}
export default Order