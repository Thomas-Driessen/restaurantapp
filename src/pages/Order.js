import React from "react";
import NavBar from "../components/Navigation bars/NavBar";
import PreviousOrders from "../components/Previous Orders List/PreviousOrders";
import CurrentOrder from "../components/Current Order/CurrentOrder";
import Grid from "@material-ui/core/Grid";
import currentFoodList from "../components/Current Order/CurrentFoodList";
import currentDrinkList from "../components/Current Order/CurrentDrinkList";
import PayForOrders from "../components/Current Order/PayForOrders";

class ViewOrder extends React.Component {
  constructor() {
    super();
    this.state = {
      tableNumber: 1,
      previousDrinks: [],
      previousFoods: [],
      pricePreviousDrinks: 0,
      pricePreviousFoods: 0,
      priceCurrentDrinks: 0,
      priceCurrentFoods: 0,
      tableId: sessionStorage.getItem("tableId"),
    };
  }

  async componentDidMount() {
    document.title = "My Order | " + this.props.name;
    let mounted = true;
    let sum = 0;
    await fetch(
      `${process.env.REACT_APP_API_URL}/api/table/tableNumber/${this.state.tableId}`
    )
      .then((res) => res.json())
      .then((data) => {
        if (mounted) {
          this.setState({ tableNumber: data.tableNumber });
        }
      })
      .catch(console.log);
    
      this.updatePreviousOrders(mounted);

    sum = currentDrinkList.reduce(
      (totalPrice, drink) => totalPrice + drink.price,
      0
    );
    this.setState({ priceCurrentDrinks: sum });
    sum = currentFoodList.reduce(
      (totalPrice, food) => totalPrice + food.price,
      0
    );
    this.setState({ priceCurrentFoods: sum });

    return () => (mounted = false);
  }

  countIdsInArray(id, array) {
    let count = 0;
    array.forEach((element) => {
      if (element.id === id) {
        count++;
      }
    });
    return count;
  }

  removeFood = (product) => {
    let index = currentFoodList.indexOf(product);
    this.setState({
      priceCurrentFoods:
        this.state.priceCurrentFoods - currentFoodList[index].price,
    });
    currentFoodList.splice(index, 1);
    sessionStorage.setItem("currentFoodList", JSON.stringify(currentFoodList));
  };

  removeDrink = (product) => {
    let index = currentDrinkList.indexOf(product);
    this.setState({
      priceCurrentDrinks:
        this.state.priceCurrentDrinks - currentDrinkList[index].price,
    });
    currentDrinkList.splice(index, 1);
    sessionStorage.setItem(
      "currentDrinkList",
      JSON.stringify(currentDrinkList)
    );
  };

  sendOrder() {
    this.postOrder();
    this.updatePreviousOrders(true);
  }

  arrayContainsFoodOrDrink() {}

  updatePreviousOrders(mounted) {
    let sum = 0;
    fetch(
      `${process.env.REACT_APP_API_URL}/api/orderdrink/${this.state.tableNumber}`
    )
      .then((res) => res.json())
      .then((data) => {
        if (mounted && data) {
          sum = data.reduce(
            (totalPrice, product) => totalPrice + product.price,
            0
          );
          let prevUniqueDrinks = [];
          data.forEach((element) => {
            if (this.countIdsInArray(element.id, prevUniqueDrinks) === 0) {
              let count = this.countIdsInArray(element.id, data);
              element.count = count;
              prevUniqueDrinks.push(element);
            }
          });
          this.setState({ pricePreviousDrinks: sum });
          this.setState({ previousDrinks: prevUniqueDrinks });        }
      })
      .catch(console.log);

    fetch(
      `${process.env.REACT_APP_API_URL}/api/orderfood/${this.state.tableNumber}`
    )
      .then((res) => res.json())
      .then((data) => {
        if (mounted && data) {
          sum = data.reduce(
            (totalPrice, product) => totalPrice + product.price,
            0
          );
          let prevUniqueFoods = [];
          data.forEach((element) => {
            if (this.countIdsInArray(element.id, prevUniqueFoods) === 0) {
              let count = this.countIdsInArray(element.id, data);
              element.count = count;
              prevUniqueFoods.push(element);
            }
          });
          this.setState({ pricePreviousFoods: sum });
          this.setState({ previousFoods: prevUniqueFoods });
        }
      })
      .catch(console.log);
  }

  postOrder() {
    let order = [];
    currentDrinkList.map((currentDrink) => {
      var drink = {
        tableId: this.state.tableNumber,
        paid: false,
        drink: {
          id: currentDrink.id
        }
      };
      fetch(`${process.env.REACT_APP_API_URL}/api/orderdrink`, {
        method: "POST",
        mode: "cors",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
        body: JSON.stringify(drink),
      });
      let item = {
        title: currentDrink.title,
      };
      order.push(item);
      return "Succes";
    });

    currentFoodList.map((currentFood) => {
      var food = {
        tableId: this.state.tableNumber,
        paid: false,
        food: {
          id: currentFood.id
        }
      };
      fetch(`${process.env.REACT_APP_API_URL}/api/orderfood`, {
        method: "POST",
        mode: "cors",
        headers: {
          Accept: "application/json",
          "Content-Type": "application/json",
        },
        body: JSON.stringify(food),
      });
      let item = {
        title: currentFood.title,
      };
      order.push(item);
      return "Succes";
    });

    currentDrinkList.splice(0, currentDrinkList.length);
    currentFoodList.splice(0, currentFoodList.length);
    sessionStorage.removeItem("currentDrinkList");
    sessionStorage.removeItem("currentFoodList");
    this.setState({ priceCurrentDrinks: 0 });
    this.setState({ priceCurrentFoods: 0 });

    fetch(`${process.env.REACT_APP_API_URL}/sendorder`, {
      method: "POST",
      mode: "cors",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
      body: JSON.stringify(order),
    });
  }

  render() {
    return (
      <div>
        <NavBar pageName="order" />
        <Grid container spacing={0} style={{ padding: 15 }}>
          <Grid item xs={12} sm={12} lg={6} xl={6}>
            <PayForOrders
              priceCurrentDrinks={this.state.priceCurrentDrinks}
              priceCurrentFoods={this.state.priceCurrentFoods}
              pricePreviousDrinks={this.state.pricePreviousDrinks}
              pricePreviousFoods={this.state.pricePreviousFoods}
              tableId={this.state.tableId}
            />
            <CurrentOrder
              totalPrice={
                this.state.priceCurrentDrinks + this.state.priceCurrentFoods
              }
              sendOrder={() => this.sendOrder()}
              removeDrink={this.removeDrink}
              removeFood={this.removeFood}
            />
          </Grid>
          <Grid item xs={12} sm={12} lg={6} xl={6}>
            <PreviousOrders
              totalPrice={
                this.state.pricePreviousDrinks + this.state.pricePreviousFoods
              }
              previousFoods={this.state.previousFoods}
              previousDrinks={this.state.previousDrinks}
            />
          </Grid>
        </Grid>
      </div>
    );
  }
}
export default ViewOrder;
