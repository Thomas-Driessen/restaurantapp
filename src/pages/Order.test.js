import React from 'react';
import { render } from '@testing-library/react';
import Order from './Order'
import renderer from 'react-test-renderer';
import { createMuiTheme } from '@material-ui/core/styles';

const theme = createMuiTheme({
  palette: {
    primary: {
      light: '#f05545',
      main: '#b71c1c',
      dark: '#7f0000',
      contrastText: '#fff',
    },
    secondary: {
      light: '#ff7961',
      main: '#f44336',
      dark: '#ba000d',
      contrastText: '#000',
    },
  },
});

it('Order renders correctly', () => {
  const tree = renderer
    .create(<Order theme={theme} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Order', () => {
  const { getByText } = render(<Order theme={theme} />);
  const linkElement = getByText(/Pay for orders/i);
  expect(linkElement).toBeInTheDocument();
});

test('Render Order', () => {
  const { getByText } = render(<Order theme={theme} />);
  const linkElement = getByText(/This is your current order/i);
  expect(linkElement).toBeInTheDocument();
});

afterAll(() => {
  global.gc && global.gc();
})