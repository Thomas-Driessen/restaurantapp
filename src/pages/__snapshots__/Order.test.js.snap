// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order renders correctly 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          text-align="center"
        >
          Restaurant
        </h6>
        <div
          className="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
        >
          <div
            style={
              Object {
                "float": "right",
              }
            }
          >
            <div
              aria-label="text primary button group"
              className="MuiButtonGroup-root"
              role="group"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Menu
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <span
                    className="MuiButton-startIcon MuiButton-iconSizeLarge"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        r="3.2"
                      />
                      <path
                        d="M9 2L7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
                      />
                    </svg>
                  </span>
                  Scan QR
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
    style={
      Object {
        "padding": 15,
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <header
          className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorTransparent MuiPaper-elevation0"
          style={
            Object {
              "paddingLeft": 15,
              "paddingRight": 20,
            }
          }
        >
          <div
            className="MuiToolbar-root MuiToolbar-regular"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "borderRadius": 50,
                  "float": "left",
                  "minHeight": "40px",
                  "minWidth": "200px",
                }
              }
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Pay for Orders
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <div
              className="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
            >
              <h3
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                Total:  
                <span
                  style={
                    Object {
                      "color": "green",
                    }
                  }
                >
                  0.00
                  €
                </span>
              </h3>
            </div>
          </div>
        </header>
      </div>
      <div>
        <h2
          style={
            Object {
              "paddingLeft": 20,
              "textDecorationColor": "primary",
              "textDecorationLine": "underline",
            }
          }
        >
          This is your current order
        </h2>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          style={
            Object {
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        />
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          style={
            Object {
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        />
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-lg-6 MuiGrid-grid-xl-6"
    >
      <div>
        <h2
          style={
            Object {
              "paddingLeft": 20,
              "textDecorationColor": "primary",
              "textDecorationLine": "underline",
            }
          }
        >
          These are your previous orders
        </h2>
        <h3
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          Foods
        </h3>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          style={
            Object {
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        />
        <h3
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          Drinks
        </h3>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          style={
            Object {
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        />
        <h3
          style={
            Object {
              "float": "right",
              "paddingRight": 20,
            }
          }
        >
          Total to be paid:  
          <span
            style={
              Object {
                "color": "green",
              }
            }
          >
            0.00
            €
          </span>
        </h3>
      </div>
    </div>
  </div>
</div>
`;
